src/components/common/select/CustomSelect.tsx

```diff
@@ -132,9 +144,59 @@ function CustomSelect<T extends string>(props: CustomSelectProps<T>) {
     return () => document.removeEventListener('mousedown', handleClickOutside);
   }, []);
 
-  const hasSelection = value !== undefined && value !== '';
-  const selectedLabel = children ? triggerLabel : options.find((option) => option.value === value)?.label || placeholder || value;
-  const isHighlight = highlightOnSelect && hasSelection;
+  const selectedLabel = useMemo(() => {
+    if (children) return triggerLabel;
+
+    if (props.isMulti) {
+      const currentValues = props.value;
+      const targetValues = currentValues.length === 0 ? options.map((opt) => opt.value) : currentValues;
+      const labels = targetValues.map((val) => options.find((opt) => opt.value === val)?.label).filter(Boolean);
+
+      return labels.length > 0 ? labels.join(', ') : placeholder;
+    }
+
+    const foundOption = options.find((option) => option.value === props.value);
+    return foundOption?.label || placeholder || props.value;
+  }, [children, triggerLabel, props, options, placeholder]);
+
+  const isHighlight = useMemo(() => {
+    if (!highlightOnSelect) return false;
+
+    if (props.isMulti) {
+      return true;
+    }
```

Comment:
```suggestion
    if (!highlightOnSelect) {
    	return false;
  	}

    if (props.isMulti) {
      return true;
    }
```

이런 indent 들어가는 구문들은 통일해야 읽기 편하더라구요~